spec/controllers/course/achievement/condition/assessments_controller_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'rails_helper'
+
+RSpec.describe Course::Achievement::Condition::AssessmentsController, type: :controller do
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let(:user) { create(:administrator) }
+    let(:course) { create(:course) }
+
+    before { sign_in(user) }
+
+    describe '#destroy' do
+      let(:assessment_condition) do
+        create(:course_condition_assessment, course: course).tap do |stub|
+          allow(stub).to receive(:destroy).and_return(false)
+        end
+      end
+      let(:achievement) do
+        create(:course_achievement,
+               course: course,
+               conditions: [assessment_condition])
+      end
+
+      subject do
+        delete :destroy,
+               course_id: course,
+               achievement_id: achievement,
+               id: assessment_condition
+      end
+
+      context 'when destroy fails' do
+        before do
+          controller.instance_variable_set(:@assessment_condition, assessment_condition)
+          controller.instance_variable_set(:@conditional, achievement)
+          subject
+        end
+
+        it 'redirects with a flash message' do
+          it { is_expected.to redirect_to(edit_course_achievement_path(course, achievement)) }
```

Comment:
it within it doesn't work...

I've fixed this in a commit after the merge, please take note.